libs/server/ACL/SecretsUtility.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+namespace Garnet.server.ACL
+{
+    internal static class SecretsUtility
+    {
+        public static bool ConstantEquals(ReadOnlySpan<byte> a, ReadOnlySpan<byte> b) => CryptographicOperations.FixedTimeEquals(a, b);
+
+        public static bool ConstantEquals(string a, string b) => ConstantEquals(a.AsSpan(), b.AsSpan());
+
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public static bool ConstantEquals(ReadOnlySpan<char> a, ReadOnlySpan<char> b)
+        {
+            // all secrets are ASCII or UTF-8
+
+            return ConstantEquals(MemoryMarshal.Cast<char, byte>(a), MemoryMarshal.Cast<char, byte>(b));
```

Comment:
I am assuming we don't care about Unicode normalization, right? (just checking)

Reply:
I scanned through all the different places encoding changes occur and it looks like they're all ASCII or UTF-8. At the moment this method isn't actually called anywhere and it's added as matter of completeness so I could

1. Do nothing
2. Remove it
3. Add a UTF-16 detection+conversion before cast
4. Re-add the original implementation that compares `char` directly
5. Throw if UTF-16

Not keen on (3) but everything else is fair game.

Thoughts?

Reply:
I don’t have any specific concern, so it was merely a question for curiosity/completeness. I guess option (1) or (2) would be fine for now.

Lot of authentication specifications impose explicit requirements for string preparation (notably, the StringPrep framework and SASLprep RFCs specifically come to mind), so I figured I would ask if any of that was a concern here if a secret is represented as a string.

Reply:
Since all plain-text secrets are currently encoded in ASCII (because passwords need to be passed through the RESP-protocol), I would suggest removing the unneeded functions for now. If we ever need constant time comparisons for UTF-16 strings, we can add back the original implementation.

Reply:
> Since all plain-text secrets are currently encoded in ASCII (because passwords need to be passed through the RESP-protocol), I would suggest removing the unneeded functions for now. If we ever need constant time comparisons for UTF-16 strings, we can add back the original implementation.

Fair enough. Let me it and I'll add some comments to indicate potential options for string conversions.